app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/deployManifestStage.ts

```diff
@@ -35,5 +38,18 @@ module(KUBERNETES_DEPLOY_MANIFEST_STAGE, [
         { type: 'requiredField', fieldName: 'moniker.cluster', fieldLabel: 'Cluster' }
       ],
     });
+
+    artifactReferenceServiceProvider.registerReference('stage', (obj: IStage, fn: any) => {
+      if (obj.type === 'deployManifest') {
```

Comment:
I wonder if `Walker` should retain the `stage` or `trigger` type or name. You could run a `Walker`s callback only if the stage or trigger matches the type of the `Walker`, rather than relying on these kinds of `if` blocks to decide if the callback is relevant to the stage.